### PR TITLE
cht-sh: init at unstable-2018-11-02

### PR DESCRIPTION
**Pull request by fgaz**
###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Assured whether relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

### CODE REVIEW

File: pkgs/top-level/all-packages.nix [hunk 1]
```diff
@@ -1140,6 +1140,8 @@ with pkgs;
 
   cfdyndns = callPackage ../applications/networking/dyndns/cfdyndns { };
 
+  cht-sh = callPackage ../tools/misc/cht.sh { };
```
Comment (fgaz) [inline pkgs/top-level/all-packages.nix]: directory name: cht.sh
package name: cht.sh
attribute name: cht-sh

is that ok?

Comment (Ma27) [reply pkgs/top-level/all-packages.nix]: I'd say yes. `cht.sh` is interpreted as `cht = { sh = <your package>; };` and this can be fairly confusing IMHO

Comment (fgaz) [reply pkgs/top-level/all-packages.nix]: Updated the commit message

File: pkgs/tools/misc/cht.sh/default.nix [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+    owner = "chubin";
+    repo = "cheat.sh";
+    rev = "9595805ac68b3c096f7c51fa024dcb97a7dfac44";
+    sha256 = "11g8say5fksg0zg0bqrgl92rprn4lwp20g9rz1i0r38f0jy3nyrf";
+  };
+
+  installPhase = ''
+    install -m755 -D share/cht.sh.txt "$out/bin/cht.sh"
+    wrapProgram "$out/bin/cht.sh" --prefix PATH : \
+    "${stdenv.lib.makeBinPath [ curl rlwrap ncurses xsel ]}"
```
Comment (c0bw3b) [inline pkgs/tools/misc/cht.sh/default.nix]: It's customary to indent the second part of a command following a backslash.

You can do:
```nix
  wrapProgram $out/bin/cht.sh --prefix PATH : \
    "${stdenv.lib.makeBinPath [ curl rlwrap ncurses xsel ]}"
```

Or:
```nix
  wrapProgram $out/bin/cht.sh \
    --prefix PATH : "${stdenv.lib.makeBinPath [ curl rlwrap ncurses xsel ]}"
```

Comment (fgaz) [reply pkgs/tools/misc/cht.sh/default.nix]: Good catch! Fixed

**Comment by worldofpeace** [comment]
@GrahamcOfBorg build cht-sh

**Review by worldofpeace — APPROVED** [review]
Tested it locally and it appears functional.

**Comment by worldofpeace** [comment]
Weird, the help menu displays
```
Usage:

    .cht.sh-wrapped --help           show this help
    .cht.sh-wrapped --shell [LANG]   shell mode (open LANG if specified)
    .cht.sh-wrapped QUERY            process QUERY and exit
```
**.cht.sh-wrapped**

**Comment by Ma27** [comment]
I couldn't get it working either. And I'm not sure about the semantics of the internally used `exec -a`: `exec -a foo bash` works perfectly fine (`$0` = `foo`), but when checking this out with scripts it doesn't work anymore.

Unless there's a simple way I'd suggest fixing the references in the help message via `substituteInPlace`.

**Comment by fgaz** [comment]
I think it's because it re-executes itself: https://github.com/chubin/cheat.sh/blob/9595805ac68b3c096f7c51fa024dcb97a7dfac44/share/cht.sh.txt#L444

**Comment by Ma27** [comment]
ah I see :)
In this case it would IMHO be best to patch `$0` invokations in `--help` with `cht.sh` :)

**Comment by fgaz** [comment]
@Ma27 Done

**Pull request merged**